install.sh

```diff
@@ -0,0 +1,405 @@
+
+  # If the current directory already contains the download, then
+  # there's no need to download it
+  if [ -f ${name} ]; then
+    echo ""
+  else
+
+    case $os in
+      "RedHat" | "CentOS")
+        echo "${CDN_URL}/centos-${ver}/pkgs/${name}"
+        ;;
+      "Ubuntu")
+        echo "${CDN_URL}/ubuntu-${ver}/pkgs/${name}"
+        ;;
+      "Debian")
+        echo "${CDN_URL}/debian-9/pkgs/${name}"
+        ;;
+      "SUSE")
+        echo "${CDN_URL}/opensuse-42/pkgs/${name}"
+        ;;
+      "SUSE15")
+        echo "${CDN_URL}/opensuse-15/pkgs/${name}"
+        ;;
+    esac
+  fi
+}
+
+# Given a version or "latest", returns a version to download. If no
+# valid input version is given, returns blank ("").
+get_version () {
+  versions=(${R_VERSIONS})
+  version_input=$1
+  if [ "${version_input}" = "latest" ]; then
+    version_input=${versions[0]}
+  fi
+  # Convert short version to real version
+  echo $(valid_version $version_input)
+}
+
+# Checks to see if a version is valid
+valid_version () {
+  ver=$1
+  result=
+  for v in ${R_VERSIONS}
+  do
+    if [[ "${v}" = "${ver}" ]]; then
+      result=${v}
+    fi
+  done
+  echo ${result}
+}
+
+# Prompts for the version until a valid version is entered.
+SELECTED_VERSION=${R_VERSION}
+prompt_version () {
+  while [ "$SELECTED_VERSION" = "" ]; do
+    echo "Available Versions"
+    show_versions
+    echo "Enter version to install: (<ENTER> for latest)"
+    read version_input
+    if [ "$version_input" = "" ]; then
+      version_input="latest"
+    fi
+    SELECTED_VERSION=$(get_version "${version_input}")
+  done
+}
+
+# Installs R
+install () {
+  installer_type=$1
+  installer_name=$2
+  os=$3
+  ver=$4
+  if [ "$installer_type" = "deb" ]; then
+    install_deb ${installer_name}
+  else
+    install_rpm ${installer_name} ${os} ${ver}
+  fi
+}
+
+# Installs R for Ubuntu/Debian
+install_deb () {
+  installer_name=$1
+  echo "Install from DEB installer ${installer_name}..."
+
+  if ! has_sudo "apt-get"; then
+    echo "Must have sudo privileges to run apt-get"
+    exit 1
+  fi
+
+  ${SUDO} apt-get install gdebi-core
+  ${SUDO} gdebi "${installer_name}"
+}
+
+# Installs R for RHEL/CentOS and SUSE
+install_rpm () {
+  installer_name=$1
+  os=$2
+  ver=$3
+  echo "User install from RPM installer ${installer_name}..."
+  install_pre "${os}" "${ver}"
+  case $os in
+    "RedHat" | "CentOS")
+      if ! has_sudo "yum"; then
+        echo "Must have sudo privileges to run yum"
+        exit 1
+      fi
+      ${SUDO} yum install "${installer_name}"
+      ;;
+    "SUSE" | "SUSE15")
+      if ! has_sudo "zypper"; then
+        echo "Must have sudo privileges to run zypper"
+        exit 1
+      fi
+      ${SUDO} zypper --no-gpg-checks install "${installer_name}"
+      ;;
+  esac
+}
+
+# Installs prerequisites for RHEL/CentOS and SUSE
+install_pre () {
+  os=$1
+  ver=$2
+
+  case $os in
+    "RedHat" | "CentOS")
+      install_epel "${ver}"
+      ;;
+    "SUSE")
+      install_sci
```

Comment:
This will install the SLES 12 Science repo on openSUSE 42 as well, which isn't necessary. Is there a way to distinguish between openSUSE/SLES?